UnitTests/Tests.cs

```diff
@@ -1027,7 +1013,7 @@ public void WriteByteUpdatesLength()
         public void SafeReadByteDoesNotUpdateStreamPosition()
         {
             var stream = this.GetRandomStream();
-            for (var i = 0; i < stream.Length; i++)
+            for (long i = 0; i < stream.Length; i++)
             {
```

Comment:
Why the change? `var` assumes the correct type.

You can keep `var` if you use the correct literal: `0L`.